pkg/v22/resource/clusterconfigmap/desired.go

```diff
@@ -27,10 +33,26 @@ func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*
 		return nil, microerror.Mask(err)
 	}
 
-	values := map[string]string{
+	// We set the number of replicas to the number of worker nodes. This is set
+	// by the workercount resource using the current number of nodes from the
+	// tenant cluster.
```

Comment:
We still don't have confidence to enable HPA for all clusters. So we keep on using the "daemonset" approach of matching the replicas to the worker nodes.

This improves the situation because we use the current number of nodes rather than desired. This helps as there is often a delay until the nodes scale.